compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MainMethodGen.java

```diff
@@ -260,11 +339,22 @@ private void invokeConfigInit(MethodVisitor mv, PackageID packageID) {
 
         mv.visitVarInsn(ASTORE, configDetailsIndex);
 
+
         mv.visitVarInsn(ALOAD, configDetailsIndex);
         mv.visitFieldInsn(GETFIELD, CONFIG_DETAILS, "paths", "[L" + PATH + ";");
         mv.visitVarInsn(ALOAD, configDetailsIndex);
         mv.visitFieldInsn(GETFIELD, CONFIG_DETAILS, "configContent", "L" + STRING_VALUE + ";");
         mv.visitMethodInsn(INVOKESTATIC, configClass, CONFIGURE_INIT, INIT_CONFIG, false);
+
+        String moduleInitClass = JvmCodeGenUtil.getModuleLevelClassName(packageID, MODULE_INIT_CLASS_NAME);
+        mv.visitFieldInsn(GETSTATIC, moduleInitClass, CURRENT_MODULE_VAR_NAME, GET_MODULE);
+        mv.visitVarInsn(ALOAD, 6);
+        mv.visitVarInsn(ALOAD, 0);
+        mv.visitVarInsn(ALOAD, configDetailsIndex);
+        mv.visitFieldInsn(GETFIELD, CONFIG_DETAILS, "paths", "[L" + PATH + ";");
```

Comment:
Shall we moved to JVMSignatures "[L" + PATH + ";". Lets fix in other places too

Reply:
Using constant will increase the performance due to string concatenations

Reply:
Fixed in 8da28c04c1397a39613ef773b0f56c5989ce3016.